src/main/java/org/rulez/demokracia/pdengine/SessionFactoryManager.java

```diff
@@ -26,5 +26,11 @@ public String getWsUserName() {
 	public boolean hasAssurance(final String role) {
 		return getWsContext().isUserInRole(role);
 	}
+	
+	public User getWsUser(final String proxyId) {
+		User user = new User(proxyId);
```

Comment:
I cant mock it nicely. So I add it. I am open for observation :)

Reply:
SessionFactoryManager.getWsUserName() is what you are looking for. It gives back the proxy id.
What we will eventually do is `new ADAAssuranceProvider().getassurancesFor(proxyId)`.
Mock or fake that.
And never in the production code.

Reply:
of course the assurance provider class will be obtained through the initial context.